src/java.base/share/classes/java/security/Security.java

```diff
@@ -262,20 +266,37 @@ static void loadInclude(String propFile) {
             }
         }
 
+        private static Path resolve(Path path) {
+            // JDK-8352728: we prefer java.io.File::getCanonicalFile over
+            // java.nio.file.Path::toRealPath because the former is more
+            // fault-tolerant, since the canonical form of a pathname is
+            // specified to exist even for nonexistent/inaccessible files.
+            try {
+                return path.toFile().getCanonicalFile().toPath();
+            } catch (IOException e) {
+                throw new InternalError("Cannot resolve path", e);
+            }
+        }
+
+        private static void checkCyclicInclude(Path path) {
+            for (Path activePath : activePaths) {
+                try {
+                    if (Files.isSameFile(path, activePath)) {
+                        throw new InternalError(
+                                "Cyclic include of '" + resolve(path) + "'");
+                    }
+                } catch (IOException ignore) {}
```

Comment:
Not sure you want to ignore this - seems better to let this propagate and be thrown as an `InternalError`.